### ISSUE TITLE
ksmbd: not respecting sticky bit permission on shared directory

### ISSUE THREAD

**Issue by riobard** [cifsd-team/ksmbd#610]
[Sticky bit](https://en.wikipedia.org/wiki/Sticky_bit) is used to implement shared directories:

> The most common use of the sticky bit is on [directories](https://en.wikipedia.org/wiki/Directory_(computing)) residing within filesystems for [Unix-like](https://en.wikipedia.org/wiki/Unix-like) operating systems. When a directory's sticky bit is set, the filesystem treats the files in such directories in a special way so only the file's owner, the directory's owner, or [root](https://en.wikipedia.org/wiki/Superuser) can rename or delete the file. Without the sticky bit set, any user with write and execute permissions for the directory can rename or delete contained files, regardless of the file's owner. Typically, this is set on the [/tmp directory](https://en.wikipedia.org/wiki/Temporary_folder) to prevent ordinary users from deleting or moving other users' files. This feature was introduced in 4.3BSD in 1986, and today it is found in most modern Unix-like systems.

`ksmbd` does not correctly handle permissions with sticky bit. Steps to demonstrate  on Alpine Linux 3.19.1 with Linux 6.6.21 kernel and ksmbd-tool version 3.5.1.

(Assuming there exist two users `user1` and `user2` whose primary group is `share`, and `ksmbd.adduser` correctly configured for both.)

```sh
sudo mkdir -p /mnt/share
cd /mnt/share
sudo chgrp share .
sudo chmod g+xw,+t .
sudo -u user1 touch file-by-user1.txt
sudo -u user2 touch file-by-user2.txt
```

`/etc/ksmbd/ksmbd.conf` config:
``` ini
[test]
path = /mnt/share
write list = @share
```

Connect to `test` over SMB as `user1`, and you can delete both `.txt` files. 

Expected behavior: deny deletion of `file-by-user2.txt` due to permission violation.

**Comment by namjaejeon** [comment]
Okay, there is no problem against samba. only ksmbd issue. right ?

**Comment by riobard** [comment]
@namjaejeon Yes, samba handles the sticky bit correctly, and only ksmbd exhibits the problem.

**Comment by Livy-Inverse** [comment]
After 2 years, the issue still persists on Linux 6.12.66 and ksmbd 3.5.6.
